### ISSUE TITLE
fhir-proxy | 9 apache vulnerabilities fixed in apache 2.4.61

### ISSUE THREAD

**Issue by christiangierschner** [datasharingframework/dsf#214]
Hi,

there are 9 apache vulnerabilities fixed in apache 2.4.61

[CVE-2024-39884](https://www.cve.org/CVERecord?id=CVE-2024-39884) important
[CVE-2024-36387](https://www.cve.org/CVERecord?id=CVE-2024-36387) low
[CVE-2024-38472](https://www.cve.org/CVERecord?id=CVE-2024-38472) important
[CVE-2024-38473](https://www.cve.org/CVERecord?id=CVE-2024-38473) moderate
[CVE-2024-38474](https://www.cve.org/CVERecord?id=CVE-2024-38474) important
[CVE-2024-38475](https://www.cve.org/CVERecord?id=CVE-2024-38475) important
[CVE-2024-38476](https://www.cve.org/CVERecord?id=CVE-2024-38476) important
[CVE-2024-38477](https://www.cve.org/CVERecord?id=CVE-2024-38477) important
[CVE-2024-39573](https://www.cve.org/CVERecord?id=CVE-2024-39573) moderate

2.4.61 is used in fhir-proxy

Could you please update apache to 2.4.61 or offer a mitigation or give an vulnerablility assessment.

thanks..

Christian

**Comment by hhund** [comment]
_TL;DR: CVE-2024-38477 could be used for denial-of-service attacks. We will release updated images shortly._

> **Apache HTTP Server: mod_rewrite proxy handler substitution CVE-2024-39573**
> Potential SSRF in mod_rewrite in Apache HTTP Server 2.4.59 and earlier allows an attacker to cause unsafe RewriteRules to unexpectedly setup URL's to be handled by mod_proxy.

`mod_rewrite` module is not loaded in the fhir_proxy and bpe_proxy images.

  > **Apache HTTP Server: Crash resulting in Denial of Service in mod_proxy via a malicious request CVE-2024-38477**
> null pointer dereference in mod_proxy in Apache HTTP Server 2.4.59 and earlier allows an attacker to crash the server via a malicious request.

`mod_proxy` module is used in the fhir_proxy and bpe_proxy images. We will release a new docker images based on Apache httpd 2.4.61 shortly. Impact: Denial of service. Recommendation: Monitor DSF availability.

> **Apache HTTP Server may use exploitable/malicious backend application output to run local handlers via internal redirect CVE-2024-38476**
> Vulnerability in core of Apache HTTP Server 2.4.59 and earlier are vulnerably to information disclosure, SSRF or local script execution via backend applications whose response headers are malicious or exploitable.
> Note: Some legacy uses of the 'AddType' directive to connect a request to a handler must be ported to 'SetHandler' after this fix.

Theoretically exploitable via response headers from DSF FHIR / BPE applications.

> **Apache HTTP Server weakness in mod_rewrite when first segment of substitution matches filesystem path. CVE-2024-38475**
> Improper escaping of output in mod_rewrite in Apache HTTP Server 2.4.59 and earlier allows an attacker to map URLs to filesystem locations that are permitted to be served by the server but are not intentionally/directly reachable by any URL, resulting in code execution or source code disclosure.
> Substitutions in server context that use a backreferences or variables as the first segment of the substitution are affected. Some unsafe RewiteRules will be broken by this change and the rewrite flag "UnsafePrefixStat" can be used to opt back in once ensuring the substitution is appropriately constrained.

`mod_rewrite` module is not loaded in the fhir_proxy and bpe_proxy images.

> **Apache HTTP Server weakness with encoded question marks in backreferences CVE-2024-38474**
> Substitution encoding issue in mod_rewrite in Apache HTTP Server 2.4.59 and earlier allows attacker to execute scripts in directories permitted by the configuration but not directly reachable by any URL or source disclosure of scripts meant to only to be executed as CGI.

`mod_rewrite` module is not loaded in the fhir_proxy and bpe_proxy images.

> **Apache HTTP Server proxy encoding problem CVE-2024-38473**
> Encoding problem in mod_proxy in Apache HTTP Server 2.4.59 and earlier allows request URLs with incorrect encoding to be sent to backend services, potentially bypassing authentication via crafted requests. This affects configurations where mechanisms other than ProxyPass/ProxyPassMatch or RewriteRule with the 'P' flag are used to configure a request to be proxied, such as SetHandler or inadvertent proxying via CVE-2024-39573. Note that these alternate mechanisms may be used within .htaccess.

`mod_proxy` module is used in the fhir_proxy and bpe_proxy images. Images only use the unaffected `ProxyPass` and `ProxyPassReverse` rules.

> **Apache HTTP Server on WIndows UNC SSRF CVE-2024-38472**
> SSRF in Apache HTTP Server on Windows allows to potentially leak NTML hashes to a malicious server via SSRF and malicious requests or content
> Users are recommended to upgrade to version 2.4.60 which fixes this issue. Note: Existing configurations that access UNC paths will have to configure new directive "UNCList" to allow access during request processing.

fhir_proxy and bpe_proxy docker images are based on Alpine Linux.

> **Apache HTTP Server: DoS by Null pointer in websocket over HTTP/2 CVE-2024-36387**
> Serving WebSocket protocol upgrades over a HTTP/2 connection could result in a Null Pointer dereference, leading to a crash of the server process, degrading performance.

`http2_module` module is not loaded in the fhir_proxy and bpe_proxy images.

> **Apache HTTP Server: source code disclosure with handlers configured via AddType CVE-2024-39884**
> A regression in the core of Apache HTTP Server 2.4.60 ignores some use of the legacy content-type based configuration of handlers. "AddType" and similar configuration, under some circumstances where files are requested indirectly, result in source code disclosure of local content. For example, PHP scripts may be served instead of interpreted.

fhir_proxy and bpe_proxy docker images are only used as reverse proxies and do not server content directly.
